quarkus.ls.ext/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/extensions/ollama/OllamaItemMetadataProvider.java

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+* Copyright (c) 2024 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.quarkus.extensions.ollama;
+
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.DEFAULT_OLLAMA_BASE_URL;
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.QUARKUS_LANGCHAIN4J_OLLAMA_BASE_URL_KEY;
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY;
+
+import java.io.StringReader;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.eclipse.lsp4mp.commons.metadata.ItemHint;
+import org.eclipse.lsp4mp.commons.metadata.ItemMetadata;
+import org.eclipse.lsp4mp.commons.metadata.ValueHint;
+import org.eclipse.lsp4mp.commons.utils.StringUtils;
+import org.eclipse.lsp4mp.extensions.ExtendedMicroProfileProjectInfo;
+import org.eclipse.lsp4mp.extensions.ItemMetadataProvider;
+import org.eclipse.lsp4mp.model.Node;
+import org.eclipse.lsp4mp.model.Node.NodeType;
+import org.eclipse.lsp4mp.model.PropertiesModel;
+import org.eclipse.lsp4mp.model.Property;
+
+import com.google.gson.GsonBuilder;
+
+/**
+ * Properties provider to collect available Ollama model values for the property
+ * 'quarkus.langchain4j.ollama.chat-model.model-id'.
+ * 
+ * The Ollama models are collected by using Ollama Search Api by using the
+ * following base url:
+ * 
+ * <ul>
+ * <li>defined with 'quarkus.langchain4j.ollama.base-url' property.
+ * <li>otherwise with 'http://localhost:11434'</li>
+ * </ul>
+ */
+public class OllamaItemMetadataProvider implements ItemMetadataProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(OllamaItemMetadataProvider.class.getName());
+
+	private static final long REFRESH_OLLAMA_CACHE_TIMEOUT = 3600L;

```

Comment:
should be 1 min. That's the time it'd take if you pull a new image into ollama on a very fast connection. You don't want to have to restart the IDE to get the new image

Reply:
fixed